### PR TITLE
migrate azure-disk and azure-file CSI addons to upstream

### PR DESCRIPTION
**Pull request by xrstf**
**What this PR does / why we need it**:
This PR replaces our hand-copied Azure CSI addons with addons created from the upstream Helm charts. We already did similar things for some other addons and now it's Azure's turn.

Similar to / inspired by KubeOne, my plan is that each addon will have its own `Makefile` in the future and that we get rid of these annoying "once you copied it from upstream, change these things: - bla bla" tasks entirely by automating everything.

For most things, kustomize seems like a good fit, but ultimatly we still need a way to wrap all Docker image references with our templating logic (`{{ Image "..." }}`). The old shell script we already have (that just runs `sed`) appeared too fragile to me and so I decided to write a tiny Go program and use our existing `yamled` package to perform these replacements. Since the addons are now fully automated, it doesn't matter that we lose formatting and comments when piping the YAML through `yamled`. Existing Makefiles continue to use the old script for the time being until someone in another PR updates those as well.

As was expected, the new manifests have different labels than the old ones and so a simple in-place upgrade is not possible. I added new migration code to the addon controller to delete the old Deployments/DaemonSets. Thanks to the new addon migration logic we just merged this code is even unit tested now :-)

I removed the `t.Parallel()` from the tests as running them in parallel has shown to be very unstable in CI.

part of #13597

**What type of PR is this?**
/kind deprecation
/kind chore

**Does this PR introduce a user-facing change? Then add your Release Note here**:
```release-note
The azuredisk/azurefile CSI addons have been replaced with manifests based on the upstream Helm chart.
```

**Documentation**:
```documentation
NONE
```

### CODE REVIEW

File: addons/csi/azure-file/Makefile [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+.PHONY: build
+build:
+	helm --namespace kube-system template azurefile-csi-driver $(REPO_NAME)/azurefile-csi-driver \
+	  --version $(CHART_VERSION) \
+	  --set 'windows.enabled=false' \
+	  --set 'controller.replicas=1' \
+	  --set 'controller.podAnnotations.cluster-autoscaler\.kubernetes\.io/safe-to-evict-local-volumes=socket-dir' \
+	  --skip-tests \
+	  > $(OUTPUT_FILE)
```
Comment (SimonTheLeg) [inline addons/csi/azure-file/Makefile:32]: q: Is there a reason why we cannot use kustomizes helm feature here? If so, I think we should document it

**Review by SimonTheLeg — COMMENTED** [review]
looks good to me. I have left one question, but it's no deal-breaker for me at all. More of a I am curious.

Otherwise: I am wondering if templatify shouldn't be it's own separate binary and we add it to the Kubermatic build image. But this tbf structuring of CI tooling is not my area of expertise. So no blocker on this as well :)

File: addons/csi/azure-file/Makefile (hunk 1 above)
Comment (xrstf) [reply addons/csi/azure-file/Makefile:32]: I consider Helm to be the source and Kustomize to be one of the transformation steps, hence why I separated the two. Personally I see no reason to intermingle Helm rendering and kustomizing in the first place, that Helm integration smells like feature creep to me. ^^

Comment (SimonTheLeg) [reply addons/csi/azure-file/Makefile:32]: hmm interesting. For me it's the opposite: I consider kustomize the source of truth, especially with a pinned helm-chart version. It's the higher-power renderer responsible for producing a predictable result.
But tbf, this is a tomatos <> tomatoes kind of thing for me. Will approve the PR now

**Pull request merged**